src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/identifier_control_form.tsx

```diff
@@ -62,23 +66,38 @@ export function FieldControlForm({
   onCancelControl,
   search,
   closeFlyout,
-}: FieldControlFormProps) {
+}: IdentifierControlFormProps) {
   const suggestedVariableName = useMemo(() => {
     const existingVariables = esqlVariables.filter((variable) => variable.type === variableType);
```

Comment:
There is always the same map below, so perhaps we can do it directly. Then you use it to make sure that the variable name is unique, hence it can be a Set instead that you pass to `getRecurrentVariableName`
```suggestion
    const existingVariables = new Set(esqlVariables.filter((variable) => variable.type === variableType).map((variable) => variable.key));
```

Reply:
Done https://github.com/elastic/kibana/pull/210170/commits/7bb3bb14142baf4fbd9820d7feb558c6a292742a